mono/mini/driver.c

```diff
@@ -1689,6 +1689,18 @@ mono_get_version_info (void)
 
 static gboolean enable_debugging;
 
+void
+handle_stats_name (char *arg)
+{
+	size_t method_name_len = strlen (arg);
+	if (method_name_len == 0) {
+		fprintf (stderr, "Missing method name in --stats command line option\n");
+		exit (1);
+	}
+	// 8 = strlen("--stats=")
+	mono_stats_set_method_name (g_strndup (&arg [8], method_name_len));
```

Comment:
As-is, this will duplicate too much since you're getting the length of the full string but then only duplicating starting at index 8. Passing only the method name makes this problem go away as well.